osm2gtfs/core/osm_connector.py

```diff
@@ -514,21 +650,22 @@ def _generate_shape(self, route_variant, query_result_set):
     def _is_valid_stop_candidate(self, stop):
         """Helper function to check if a stop candidate has a valid tagging
 
-        Returns True or False
+        :return bool: Returns True or False
 
         """
+        valid = False
         if 'public_transport' in stop.tags:
             if stop.tags['public_transport'] == 'platform':
-                return True
+                valid = True
             elif stop.tags['public_transport'] == 'station':
-                return True
-        elif 'highway' in stop.tags:
+                valid = True
+        if 'highway' in stop.tags:
             if stop.tags['highway'] == 'bus_stop':
-                return True
-        elif 'amenity' in stop.tags:
+                valid = True
+        if 'amenity' in stop.tags:
             if stop.tags['amenity'] == 'bus_station':
-                return True
-        return False
+                valid = True
+        return valid
```

Comment:
I though it would be more elegant to fix one special case;

` if 'public_transport' in stop.tags:`
[...]
```
  else:
     return True
```

Is is better to change all `valid = ` to `return `? And get rid of the first initialization line `valid = False` and and the special "else case" as described at the beginning of this very comment? I'm happy to get get help here on how to write it best in Python (sorry, still my very first Python project).

Reply:
I don't understand the special case. Wasn't it handled properly before?

After all, it is a matter of style. I would just propose to follow the principle: I ain't broken, then don't fix it. Or: Only make changes that are necessary.

Reply:
Actually it was also handled. Yes, it is a matter of style, I prefer to have a clear flow instead of break-outs. But this doesn't have to be part of this PR. So I roll this back.